app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,49 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, only: [:new, :create, :edit, :update, :destroy]
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @user = current_user
```

Comment:
Instead of creating `@user` multiple times, why not just reference `current_user` in the view? That'll save you some lines of code in your controller.